### ISSUE TITLE
encoding/gob: debug function prints uint values incorrectly

### ISSUE THREAD

**Issue by drosseau** [golang/go#21392]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?
```
$ go version
go version go1.8 linux/amd64
```

### What operating system and processor architecture are you using (`go env`)?
```
$ go env
GOARCH="amd64"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build871485140=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
PKG_CONFIG="pkg-config"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
```

### What did you do?

Use the `Debug` function in `encoding/gob` to try to view a gob. A simplified gob that has the error is here: 
```go
package main

import (
	"bytes"
	"encoding/gob"
)

var gobBytes = []byte{0x3, 0x6, 0x0, 0xa}


func main() {
	gob.Debug(bytes.NewReader(gobBytes))
}
```

(Sorry I can't put this in the playground as you need to build the gob package without the build guard for `Debug`)
### What did you expect to see?

Output of Debug should have been

```
Start of debugging
10
```

### What did you see instead?

```
Start of debugging
5
```

Basically this was being treated as an integer and not an unsigned integer and so decoded with the

```go
y := (x >> 1)
if x&1 != 0 {
   y = ^y
}
```

integer decoding method.

**Comment by ALTree** [comment]
Looks like a copy-paste error here:

```
    case tInt:
        x := deb.int64()
        fmt.Fprintf(os.Stderr, "%s%d\n", indent, x)
    case tUint:
        x := deb.int64()
        fmt.Fprintf(os.Stderr, "%s%d\n", indent, x)
```

Notice how the `case tUint` branch calls `x := deb.int64()` instead of `x := deb.uint64()`.

**Comment by drosseau** [comment]
I submitted a fix this.
